lib/backends/cpu/op-resolve-rules.ts

```diff
@@ -0,0 +1,91 @@
+  ['Clip', '', '6+', () => new unaryOps.CpuUnaryOp(FLOAT_TYPES, unaryOps.clip)],
+  ['Concat', '', '4+', () => new CpuConcat()],
+  ['Conv', '', '1+', () => new CpuConv()],
+  ['Cos', '', '7+', () => new unaryOps.CpuUnaryOp(NUMBER_TYPES, unaryOps.cos)],
+  ['Div', '', '7+', () => new CpuBinaryOp(NUMBER_TYPES, (e1, e2) => (e1 / e2))],
+  ['Dropout', '', '7+', () => new CpuDropout()],
+  ['Elu', '', '6+', () => new unaryOps.CpuUnaryOp(FLOAT_TYPES, unaryOps.elu)],
+  ['Exp', '', '6+', () => new unaryOps.CpuUnaryOp(NUMBER_TYPES, unaryOps.exp)],
+  ['Flatten', '', '1+', () => new CpuFlatten()],
+  ['Floor', '', '6+', () => new unaryOps.CpuUnaryOp(NUMBER_TYPES, unaryOps.floor)],
+  ['Gather', '', '1+', () => new CpuGather()],
+  ['Gemm', '', '7+', () => new CpuGemm()],
+  ['GlobalAveragePool', '', '1+', () => new CpuGlobalAveragePool()],
+  ['GlobalMaxPool', '', '1+', () => new CpuGlobalMaxPool()],
+  ['ImageScaler', '', '1+', () => new CpuImageScaler()],
+  ['InstanceNormalization', '', '6+', () => new CpuInstanceNormalization()],
+  ['LeakyRelu', '', '6+', () => new unaryOps.CpuUnaryOp(FLOAT_TYPES, unaryOps.leakyRelu)],
+  ['Log', '', '6+', () => new unaryOps.CpuUnaryOp(NUMBER_TYPES, unaryOps.log)],
+  ['LRN', '', '1+', () => new CpuLrn()],
+  ['MatMul', '', '1+', () => new CpuMatMul()],
+  ['MaxPool', '', '1+', () => new CpuMaxPool()],  // TODO: support new attributes for MaxPool-8 and MaxPool-10
+  ['Mul', '', '7+', () => new CpuBinaryOp(NUMBER_TYPES, (e1, e2) => (e1 * e2))],
+  ['Neg', '', '6+', () => new unaryOps.CpuUnaryOp(NUMBER_TYPES, unaryOps.neg)],
+  ['Or', '', '7+', () => new CpuBinaryOp(['bool'], (e1, e2) => (e1 || e2))],
+  ['PRelu', '', '7+', () => new CpuBinaryOp(NUMBER_TYPES, (e1, e2) => (e1 >= 0 ? e1 : e1 * e2))],
+  ['ReduceLogSum', '', '1+', () => new cpuReduce.CpuReduceLogSum()],
+  ['ReduceMax', '', '1+', () => new cpuReduce.CpuReduceMax()],
+  ['ReduceMean', '', '1+', () => new cpuReduce.CpuReduceMean()],
+  ['ReduceMin', '', '1+', () => new cpuReduce.CpuReduceMin()],
+  ['ReduceProd', '', '1+', () => new cpuReduce.CpuReduceProd()],
+  ['ReduceSum', '', '1+', () => new cpuReduce.CpuReduceSum()],
+  ['ReduceSumSquare', '', '1+', () => new cpuReduce.CpuReduceSumSquare()],
+  ['Relu', '', '6+', () => new unaryOps.CpuUnaryOp(NUMBER_TYPES, unaryOps.relu)],
+  ['Reshape', '', '5+', () => new CpuReshape()],
+  ['Sigmoid', '', '6+', () => new unaryOps.CpuUnaryOp(NUMBER_TYPES, unaryOps.sigmoid)],
+  ['Sin', '', '7+', () => new unaryOps.CpuUnaryOp(NUMBER_TYPES, unaryOps.sin)],
+  ['Slice', '', '1+', () => new CpuSlice()],
+  ['Softmax', '', '1+', () => new CpuSoftmax()],
+  ['Sqrt', '', '6+', () => new unaryOps.CpuUnaryOp(NUMBER_TYPES, unaryOps.sqrt)],
+  ['Squeeze', '', '1+', () => new CpuSqueeze()],
+  ['Sub', '', '7+', () => new CpuBinaryOp(NUMBER_TYPES, (e1, e2) => (e1 - e2))],
+  ['Sum', '', '6+', () => new CpuSum()],  // TODO: support multidirectional broadcast for Sum-8
```

Comment:
Do we need to throw exception when it looks like broadcasting support maybe required in the implementation (since it support 6+) ?

Reply:
the exception is thrown anyway. currently it will be thrown due to checkInput failure. the only difference is the error message